plugins/CoreAdminHome/Commands/DeleteLogs.php

```diff
@@ -0,0 +1,186 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\CoreAdminHome\Commands;
+
+use Piwik\Common;
+use Piwik\Container\StaticContainer;
+use Piwik\DataAccess\RawLogDao;
+use Piwik\Date;
+use Piwik\Db;
+use Piwik\LogDeleter;
+use Piwik\Plugin\ConsoleCommand;
+use Piwik\Site;
+use Piwik\Timer;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+
+/**
+ * Command to selectively delete visits.
+ */
+class DeleteLogs extends ConsoleCommand
+{
+    private static $logTables = array(
+        'log_visit',
+        'log_link_visit_action',
+        'log_conversion',
+        'log_conversion_item',
+        'log_action'
+    );
+
+    /**
+     * @var RawLogDao
+     */
+    private $rawLogDao;
+
+    /**
+     * @var LogDeleter
+     */
+    private $logDeleter;
+
+    public function __construct(LogDeleter $logDeleter = null, RawLogDao $rawLogDao = null)
+    {
+        parent::__construct();
+
+        $this->logDeleter = $logDeleter ?: StaticContainer::get('Piwik\LogDeleter');
+        $this->rawLogDao = $rawLogDao ?: StaticContainer::get('Piwik\DataAccess\RawLogDao');
+    }
+
+    protected function configure()
+    {
+        $this->setName('logs:delete');
```

Comment:
Actually would like to rename it `core:delete-logs-data` unless someone has a better idea :)